### DISCUSSION TITLE
One-to-many create error with "Unique constraint failed on the [Id] field"

### DISCUSSION THREAD

**Discussion by Exslims** [prisma/prisma#17548] [Q&A] [answered]
Hey guys, I have the following schema
```
model Wallet {
  walletId     Int              @id @default(autoincrement())
  title        String
  periodStart  Int              @default(1) @db.SmallInt()
  categories   Category[]
}
model Category {
  categoryId  Int           @id @default(autoincrement())
  title          String        @db.VarChar(30)
  wallet      Wallet        @relation(fields: [walletId], references: [walletId])
  walletId    Int
}
```

And trying to implement simple create wallet operation with this code: 
```
  async create(createWalletDto: CreateWalletDto) {
    return this.prisma.wallet.create({
      data: {
        title: createWalletDto.title,
        periodStart: createWalletDto.periodStart,
        categories: {
          create: [
            {
              title: 'Test',
            },
          ],
        },
      },
    });
```

But got an error
```
  15 async create(userId: number, createWalletDto: CreateWalletDto) {
→ 16   return this.prisma.wallet.create(
Unique constraint failed on the fields: (`categoryId`)
```

Looks like it tried to create new categories with existing ids in DB (there are already existed categories with 1/2/3/ id in my db), but I thought it should create new categories with new ids(incremented from last in table) and assign them to new wallet, whats wrong?
Thanks in advance!

**Comment by Exslims** [comment, marked as answer]
Ok, i found what was an issue. I did a seed db with script where I explicitly indicated ids of entities. But it breaks `@autoincrement` functionality which is under the hood creates new serial sequence.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
